### PR TITLE
Fix warning: unused variable has_custom_args in aot-compiler.c:arch_init

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

File: mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -1103,6 +1102,10 @@ arch_init (MonoAotCompile *acfg)
 	acfg->llvm_label_prefix = "";
 	acfg->user_symbol_prefix = "";
 
+#if TARGET_X86 || TARGET_AMD64
```
Comment (lewurm) [inline mono/mini/aot-compiler.c]: we usually write
```c
#if defined(TARGET_X86) || defined(TARGET_AMD64)
```

Comment (jaykrell) [reply mono/mini/aot-compiler.c]: Maybe start using a newer terser equivalent form?
They are defined to be 1 and undefined values evaluate to 0.
In future hopefully `#if` changed to `if`, so we have multi-targeting AOT compiler.

Comment (lewurm) [reply mono/mini/aot-compiler.c]: not disagreeing with you, but I prefer consistency across the code base

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**